proposed/2022/dotnet-nuget-why-proposal.md

```diff
@@ -0,0 +1,139 @@
+
+Project 'projectNameB' has the following dependency graph for 'packageA'
+   [net6.0]:
+      Microsoft.ML (1.1.0) -> Microsoft.ML.Util (1.1.0) -> packageA (1.1.0)
+   [net472]:
+      Microsoft.ML (1.1.0) -> Microsoft.ML.Util (1.1.0) -> packageA (1.1.0)
+```
+
+- List dependency graph of a package given `package id` when there is a diamond dependency (a package is brought in by more than one path).
+
+```
+dotnet nuget why packageA
+
+Project 'projectNameA' has the following dependency graph for 'packageA'
+   [net6.0]:
+      Microsoft.ML (1.0.0) -> Microsoft.ML.Util (1.0.0) -> packageA (1.0.0)
+      Microsoft.ML (1.0.0) -> Microsoft.ML.SampleUtils (1.0.0) -> packageA (1.0.0)
+```
+
+- List dependency graph of a package given `package id` and `target framework`.
+
+```
+dotnet nuget why packageA -f net6.0
+
+Project 'projectNameA' has the following dependency graph for 'packageA'
+   [net6.0]:
+      Microsoft.ML (1.0.0) -> Microsoft.ML.Util (1.0.0) -> packageA (1.0.0)
+
+Project 'projectNameB' has the following dependency graph for 'packageA'
+   [net6.0]:
+      Microsoft.ML (1.1.0) -> Microsoft.ML.Util (1.1.0) -> packageA (1.1.0)
+```
+
+## Limitations 
+
+The NuGet restore operation downloads other packages during dependency resolution which are not part of the final dependency graph that the `dotnet nuget why` command prints out. The absence of a package id and version in the `project.assets.json` file indicates that a package was downloaded during dependency resolution but is not part of the final dependency graph.
+
+## Rationale and Alternatives
+
+We could have considered a minor tweak to the existing experience of “dotnet list package --include-transitive” to provide the user with a sense of a package's dependencies. This could have been a new column next to “Resolved” called “Transitively Referenced” that had a list of the top-level packages that requested the dependency. The corresponding tracking issue can be found here: https://github.com/NuGet/Home/issues/11625. Unfortunately, this solution doesn't provide an easy way to identify the dependency graph `(starting from project -> top-level package -> transitive-dependency (1) -> ..-> transitive dependency (n))` for a particular package.
+
+![](../../meta/resources/TransitiveDependencies/DotNetCLI.png)
+
+## Prior Art
+
+Within Visual Studio, there is a new panel in the Package Manager UI called “Transitive Packages” in which all transitive packages are displayed to the user. There is also an additional header titled “Top-level Packages”.
+
+When a user highlights a transitive package, they will see a pop-up that displays what top-level package(s) are bringing it in.
+
+![](../../meta/resources/TransitiveDependencies/TransitiveVSPMUI.png)
+
+However, this still does not provide the user with enough detail about how transitive packages originate, therefore we propose the `dotnet nuget why` command.
+
+## Additional improvements
```

Comment:
Great improvements list. While they all can add value, the spec is in a good state to finalize the first iteration in my opinion!